singlestone/sugar
# Update existing test with snapshots

Added snapshot test coverage to existing test. Based on per review, will add additional test as needed.

## Description



### Adding vite.config.ts
A `vite.config.ts` have been added to the following directories (see below). This change was made so that the `.snap` files can coincide at the same level of each test.
- `sugar/sugar`
- `sugar/tailwind-helpers/vite.config.ts`
- `sugar/tailwind-plugin-buttons`
- `sugar/tailwind-plugin-form`

There is likely a cleaner way of handling this by extending a higher level `vite.config.ts` file but I haven't found anything on this quite yet.

### Adding snapshots

[sugar/sugar/test/index.test.ts](https://github.com/singlestone/sugar/pull/159/files#diff-7376bd5e00dac2f27b3ca493095c266f7f50714287d4aa30258b0cfc3777847f) assertion was change to use `toMatchInlineSnapshot()` in favor of `toEqual()` to align with using the snapshot approach for testing.

Each test in [sugar/sugar/tailwind-helpers/test](https://github.com/singlestone/sugar/tree/feature/add-initial-vitests-snapshots/sugar/tailwind-helpers/test) have been updated to use `toMatchSnapshot()` instead of `toEqual()`. As such, the expected value in each test has been removed along with its assertion. The assertion is now derived from the snapshot image for each respective test. 

As an example, the [sugar/tailwind-helpers/test/color-helpers.test.ts](https://github.com/singlestone/sugar/pull/159/files#diff-4a48da99d0be443a08e24ee59cc279ad2d683f381978a784aef8cc03193f9207) no longer declares the assertion as part of the constants `noAlphaTests` and `alphaTestsCases`. The assertions are effectively maintained now by [sugar/sugar/tailwind-helpers/test/color-helpers.tests.ts.snap](https://github.com/singlestone/sugar/blob/feature/add-initial-vitests-snapshots/sugar/tailwind-helpers/test/color-helpers.test.ts.snap). In the below screenshot , the alphaTestCase for the first array value has been changed from `#031cfc` to `#099cfc`. When the test run after this change via `pnpm test` from the root of the project, the following messaging is received.

![image](https://user-images.githubusercontent.com/24687048/212772723-26b43d2e-5c81-4918-98ab-3a921e5de9fa.png)

This will inform the developer that a change has been made and to action on it accordingly (by updating the application code, test and/or snapshots).

This pattern can also be found in the `sugar/tailwind-plugin-buttons` and `sugar/tailwind-plugin-form` test files as well.

## Related Issue






[ISSUE-160](https://github.com/singlestone/sugar/issues/160)

## Motivation and Context


Pending on perspective, this change may not be required but it would solve the potential future problem of tracking changes in the application along with its expected behavior.

## How Has This Been Tested?





- Run `pnpm install` at the root of the directory
- Navigate to a test and modify its test value
- Save the changes
- Verify that the command line results track the changes and mark the test which are obsolete

## Screenshots (if appropriate)

## Types of changes




## Checklist




- [ ] I added a changeset via `yarn changeset` if my changes should be released.
